src/main/java/blusunrize/immersiveengineering/common/data/Recipes.java

```diff
@@ -107,77 +112,556 @@ protected void registerRecipes(@Nonnull Consumer<IFinishedRecipe> out)
+			.func_218643_a("has_hempcrete", this.hasItem(StoneDecoration.hempcrete))
+			.func_218647_a(out, toRL("hempcrete_slab_from_hempcrete_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.hempcrete), StoneDecoration.hempcreteStairs)
+			.func_218643_a("has_hempcrete", this.hasItem(StoneDecoration.hempcrete))
+			.func_218647_a(out, toRL("hempcrete_stairs_from_hempcrete_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.concrete), IEBlocks.toSlab.get(StoneDecoration.concrete))
+			.func_218643_a("has_concrete", this.hasItem(StoneDecoration.concrete))
+			.func_218647_a(out, toRL("concrete_slab_from_concrete_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.concrete), StoneDecoration.concreteStairs[0])
+			.func_218643_a("has_concrete", this.hasItem(StoneDecoration.concrete))
+			.func_218647_a(out, toRL("concrete_stairs_from_concrete_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.concreteTile), IEBlocks.toSlab.get(StoneDecoration.concreteTile))
+			.func_218643_a("has_concrete", this.hasItem(StoneDecoration.concrete))
+			.func_218647_a(out, toRL("concrete_tile_slab_from_concrete_tile_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.concreteTile), StoneDecoration.concreteStairs[1])
+			.func_218643_a("has_concrete", this.hasItem(StoneDecoration.concrete))
+			.func_218647_a(out, toRL("concrete_tile_stairs_from_concrete_tile_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.concreteLeaded), IEBlocks.toSlab.get(StoneDecoration.concreteLeaded))
+			.func_218643_a("has_concrete", this.hasItem(StoneDecoration.concrete))
+			.func_218647_a(out, toRL("concrete_leaded_slab_from_concrete_leaded_stonecutting"));
+		SingleItemRecipeBuilder.func_218648_a(Ingredient.fromItems(StoneDecoration.concreteLeaded), StoneDecoration.concreteStairs[2])
+			.func_218643_a("has_concrete", this.hasItem(StoneDecoration.concrete))
+			.func_218647_a(out, toRL("concrete_leaded_stairs_from_concrete_leaded_stonecutting"));
 
+		ShapedRecipeBuilder.shapedRecipe(WoodenDecoration.treatedScaffolding, 6)
+			.patternLine("iii")
+			.patternLine(" s ")
+			.patternLine("s s")
+			.key('i', IETags.getItemTag(IETags.treatedWood))
+			.key('s', IETags.treatedStick)
+			.addCriterion("has_treated_planks", hasItem(IETags.getItemTag(IETags.treatedWood)))
+			.build(out);
+		ShapedRecipeBuilder.shapedRecipe(MetalDecoration.aluScaffolding.get(MetalScaffoldingType.STANDARD), 6)
+			.patternLine("iii")
+			.patternLine(" s ")
+			.patternLine("s s")
+			.key('i', IETags.getTagsFor(EnumMetals.ALUMINUM).ingot)
+			.key('s', IETags.aluminumRod)
+			.addCriterion("has_"+toPath(MetalDecoration.aluScaffolding.get(MetalScaffoldingType.STANDARD)), hasItem(MetalDecoration.aluScaffolding.get(MetalScaffoldingType.STANDARD)))
```

Comment:
Shouldn't this be 2 calls to `hasCriterion` for alu ingots and rods? Unlocking recipes with the output item seems strange.

Reply:
Oh whops, meant to change it befor commiting but forgot. Gonna work on that now.